tasks/burrows-wheeler-transform/src/main.rs

```diff
@@ -0,0 +1,115 @@
+use core::cmp::Ordering;
+
+const STX: char = '\u{0002}';
+const ETX: char = '\u{0003}';
+
+pub fn special_cmp(lhs: &str, rhs: &str) -> Ordering {
+    let mut iter1 = lhs.chars();
+    let mut iter2 = rhs.chars();
+
+    loop {
+        match (iter1.next(), iter2.next()) {
+            (Some(lhs), Some(rhs)) => {
+                if lhs != rhs {
+                    let is_lhs_alnum = lhs.is_alphanumeric();
+                    let is_rhs_alnum = rhs.is_alphanumeric();
+
+                    let result = if is_lhs_alnum == is_rhs_alnum {
+                        lhs.cmp(&rhs)
+                    } else if is_lhs_alnum {
+                        Ordering::Less
+                    } else {
+                        Ordering::Greater
+                    };
+
+                    return result;
+                }
+            }
+            (Some(_), None) => return Ordering::Greater,
+            (None, Some(_)) => return Ordering::Less,
+            (None, None) => return lhs.cmp(&rhs),
+        }
+    }
+}
+
+fn burrows_wheeler_transform(input: &str) -> String {
+    let mut table: Vec<String> = vec![];
+
+    // add markers for the start and end
+    let input_string = format!("{}{}{}", STX, input, ETX);
+
+    // create all possible rotations
+    for (i, _) in input_string.char_indices() {
+        table.push(format!(
+            "{}{}",
+            &input_string[input_string.len() - 1 - i..],
+            &input_string[0..input_string.len() - 1 - i]
+        ));
+    }
+
+    // sort rows alphabetically
+    table.sort_unstable_by(|lhs, rhs| special_cmp(&lhs, &rhs));
+
+    // return the last column
+    table
+        .iter()
+        .map(|s| s.chars().nth_back(0).unwrap())
+        .collect::<Vec<char>>()
```

Comment:
This collect seems unnecessary. Can't you just collect into a `String` directly?